controlplane/kubeadm/controllers/controller.go

```diff
@@ -277,48 +277,50 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, cluster *
 		return ctrl.Result{}, nil
 	}
 
-	controlPlane := internal.NewControlPlane(cluster, kcp, ownedMachines)
+	controlPlane := internal.NewControlPlane(cluster, kcp, ownedMachines, logger)
+	if controlPlane.HasOperationInProgress() {
+		return ctrl.Result{}, nil
+	}
+
+	if controlPlane.NeedsInitialization() {
```

Comment:
Do we need to wait until HasOperationInProgress() returns false to initialize control plane? Is there a scenario calling initialize before doing this check to bring up the control plane faster fails?